.github/workflows/applications.yml

```diff
@@ -0,0 +1,246 @@
+      with:
+         ghc-version: ${{ matrix.ghc }}
+         cabal-version: ${{ matrix.cabal }}
+    - name: Confirm GHC and Cabal installation
+      run: |
+        ghc --version
+        cabal --version
+
+    # Project Setup
+    - name: Create cabal.project.local
+      shell: bash
+      run: |
+        cat > cabal.project.local <<EOF
+        documentation: False
+        package pact
+          tests: True
+          benchmarks: True
+          documentation: False
+          optimization: 1
+          flags: ${{ matrix.flags }} +cryptonite-ed25519
+        extra-include-dirs:
+          /opt/local/include
+          /usr/local/opt/openssl/include
+        extra-lib-dirs:
+          /opt/local/lib
+          /usr/local/opt/openssl/lib/
+        EOF
+    - name: Extend cabal.project.local for GHC-9.0.2
+      if: "startsWith(matrix.ghc, '9')"
+      shell: bash
+      run: |
+        cat >> cabal.project.local <<EOF
+        package pact
+          ghc-options: -Wwarn -Wunused-packages
+        EOF
+    - name: Add check for unused packages
+      shell: bash
+      run: |
+        cat >> cabal.project.local <<EOF
+        package pact
+          ghc-options: -Wunused-packages
+        EOF
+    - name: Print cabal.project.local
+      shell: bash
+      run: cat cabal.project.local
+    - uses: actions/cache@v3
+      name: Cache dist-newstyle
+      with:
+        path: |
+          ~/.cabal/packages
+          ~/.cabal/store
+          dist-newstyle
+        key: ${{ matrix.os }}-${{ matrix.ghc }}-4-cabal
+
+    # Build
+    - name: Update package database
+      shell: bash
+      run: cabal update
+    - name: Display outdated packages
+      run: cabal outdated
+    - name: Configure build
+      run: |
+        cabal build --dry-run
+        cabal freeze
+    - name: Sync with cabal cache
+      if: matrix.cabalcache == 'true'
+      uses: larskuhtz/cabal-cache-action@4b537195b33898fcd9adc62cee2a44986fd7b1b6
+      with:
+        bucket: "kadena-cabal-cache"
+        region: "us-east-1"
+        folder: "packages/${{ matrix.os }}"
+        aws_access_key_id: "${{ secrets.kadena_cabal_cache_aws_access_key_id }}"
+        aws_secret_access_key: "${{ secrets.kadena_cabal_cache_aws_secret_access_key }}"
+        
+    - name: Build dependencies
+      shell: bash
+      run: cabal build --only-dependencies
+    - name: Build
+      shell: bash
+      run: cabal build
+    - name: Test
+      shell: bash
+      run: cabal run tests
+
+    - name: Verify Binary Linking
+      shell: bash
+      if: "!contains(matrix.flags, '-build-tool')"
+      run: cabal run exe:pact -- --version
+
+    # Publish Artifacts
+    - name: Prepare artifacts
+      if: "!contains(matrix.flags, '-build-tool')"
+      shell: bash
+      run: |
+        export VER=$(grep '^version' pact-tng.cabal | sed -e 's/.*: *//')
+        mkdir -p artifacts/pact
+        cp $(cabal list-bin pact) artifacts/pact
+        cp CHANGELOG.md artifacts/pact
```

Comment:
We dont have one currently, not sure if this breaks CI

Reply:
There's a blank in there now but the release generates a changelog from PR names.